### PR TITLE
Message detail screen (SwiftUI)

### PR DESCRIPTION
**Pull request by RyanRory**

### CODE REVIEW

**Review by mpretty-cyro — CHANGES_REQUESTED** [review]
Code looks fine but I ran into a few issues while testing:

- [ ] When changing the image to full screen the nav section is black **(Screenshot 1)**
- [x] The push transition starts immediately when pressing the 'Info' action - should wait until after the animation moving the snapshot back completes **(video 1)**
- [ ] Link preview content doesn't render **(Video 2)** & **(Video 3)**
- [ ] Long messages break the UI **(Screenshot 2)**, **(Video 3)** & **(Video 4)**
- [ ] Excessively long messages cause the app to hang when popping from the detail screen **(Video 4)**
- [ ] App crashes when pressing the 'Info' action for voice messages


**(Screenshot 1)**


**(Screenshot 2)**



**(Video 1)**
https://github.com/oxen-io/session-ios/assets/15862619/f6b95976-98c6-469d-91d4-592d3395c37f

**(Video 2)**
https://github.com/oxen-io/session-ios/assets/15862619/caa8c3ce-77a4-483a-8ccd-ab7b6854cc45

**(Video 3)**
https://github.com/oxen-io/session-ios/assets/15862619/02b82714-0817-4958-8826-2449ac9aae56

**(Video 4)**
https://github.com/oxen-io/session-ios/assets/15862619/899306ad-ae45-457c-864a-8ad9ee1699f8

File: Session/Conversations/Message Cells/Content Views/SwiftUI/VoiceMessageView_SwiftUI.swift [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            }
+            .padding(.all, Values.smallSpacing)
+        }
+        .frame(
+            width: Self.width
+        )
+    }
+}
+
+#Preview {
```
Comment (mpretty-cyro) [inline Session/Conversations/Message Cells/Content Views/SwiftUI/VoiceMessageView_SwiftUI.swift:84]: @RyanRory Do you know if `#Preview` is limited to specific version of iOS or XCode? (looks like the CI is failing due to this and I'm wondering if we can wrap it in an `#if @available` or some other pre-processor macro)

File: Session/Shared/SessionCarouselView+SwiftUI.swift [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+    var body: some View {
+        ZStack {
+            Capsule()
+                .foregroundColor(.init(white: 0, opacity: 0.4))
+            HStack(spacing: 4) {
+                ForEach(0...maxIndex, id: \.self) { index in
+                    Circle()
+                        .fill(index == ((self.index - 1) % (self.maxIndex + 1)) ? Color.white : Color.gray)
+                        .frame(width: 6.62, height: 6.62)
```
[the remark this answers is not in the document]
Comment (mpretty-cyro) [reply Session/Shared/SessionCarouselView+SwiftUI.swift:187]: @RyanRory not sure whether you had a chance to check this or not?

**Review by mpretty-cyro — CHANGES_REQUESTED** [review]

**Review by mpretty-cyro — CHANGES_REQUESTED** [review]
- [x] Looks like quoted messages are getting their content truncated at 2 lines

**(screenshot 1)**
Screenshot 2024-03-04 at 11 03 48 am

File: Session/Conversations/Message Cells/Content Views/SwiftUI/VoiceMessageView_SwiftUI.swift (hunk 1 above)
Comment (RyanRory) [reply Session/Conversations/Message Cells/Content Views/SwiftUI/VoiceMessageView_SwiftUI.swift:84]: I think it is a Xcode 15 thing. I'll change it.

File: Session/Shared/SessionCarouselView+SwiftUI.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (RyanRory) [reply Session/Shared/SessionCarouselView+SwiftUI.swift:187]: Ah, nah, I'll check it. It might be good to make it just 7*7

**Review by mpretty-cyro — APPROVED** [review]
Looks good - the failing tests are just due to the Disappearing Messages Feature flag

**Pull request merged**